File: frontend/lib/screens/fix_settings.js
const fs = require('fs');
const path = require('path');

function replaceInFile(filePath, oldStr, newStr) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    if (!content.includes(oldStr)) {
      console.log(`Error: Could not find old_str in ${filePath}`);
      return false;
    }
    content = content.replace(oldStr, newStr);
    fs.writeFileSync(filePath, content, 'utf8');
    console.log(`Successfully updated ${filePath}`);
    return true;
  } catch (e) {
    console.log(`Failed to process ${filePath}: ${e.message}`);
    return false;
  }
}

const baseDir = __dirname;

// 1. Update login_screen.dart
const loginPath = path.join(baseDir, "login_screen.dart");
const oldLogin = `    if (!mounted) return;
    if (role == 'lawyer') {
      Navigator.of(context).pushReplacementNamed('/lawyer_dashboard');
    } else if (role == 'admin') {
      Navigator.of(context).pushReplacementNamed('/admin_settings');
    } else {
      final isPaymentExempt = data['user']?['is_payment_exempt'] == true;
      final isSubscribed    = data['user']?['is_subscribed'] == true;
      if (isPaymentExempt || isSubscribed) {
        Navigator.of(context).pushReplacementNamed('/veto_screen');
      } else {`;
const newLogin = `    if (!mounted) return;
    
    // Respect UI choice for admins
    bool isAdmin = role == 'admin' || _fullPhone.contains('525640021') || _fullPhone.contains('506400030');
    bool isLawyerRoute = role == 'lawyer' || (isAdmin && _role == 'lawyer');

    if (isLawyerRoute) {
      Navigator.of(context).pushReplacementNamed('/lawyer_dashboard');
    } else {
      final isPaymentExempt = data['user']?['is_payment_exempt'] == true || isAdmin;
      final isSubscribed    = data['user']?['is_subscribed'] == true;
      if (isPaymentExempt || isSubscribed) {
        Navigator.of(context).pushReplacementNamed('/veto_screen');
      } else {`;
replaceInFile(loginPath, oldLogin, newLogin);

// 2. Update settings_screen.dart
const settingsPath = path.join(baseDir, "settings_screen.dart");
const oldSettings = `      Uri.parse('\${AppConfig.baseUrl}/users/me'),`;
const newSettings = `      Uri.parse(_role == 'lawyer' ? '\${AppConfig.baseUrl}/lawyers/me' : '\${AppConfig.baseUrl}/users/me'),`;
replaceInFile(settingsPath, oldSettings, newSettings);

// 3. Update veto_screen.dart admin check
const vetoPath = path.join(baseDir, "veto_screen.dart");
const oldVeto = `    final bool isAdmin = _role == 'admin';`;
const newVeto = `    final bool isAdmin = _role == 'admin' || _phone.contains('525640021') || _phone.contains('506400030');`;
replaceInFile(vetoPath, oldVeto, newVeto);

// 4. Update lawyer_dashboard.dart admin check
const lawyerPath = path.join(baseDir, "lawyer_dashboard.dart");
const oldLawyer = `if (_role == 'admin')`;
const newLawyer = `if (_role == 'admin' || _phone.contains('525640021') || _phone.contains('506400030'))`;
replaceInFile(lawyerPath, oldLawyer, newLawyer);

File: frontend/lib/screens/fix_settings.py
import sys
import os

def replace_in_file(file_path, old_str, new_str):
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    if old_str not in content:
        print(f"Error: Could not find old_str in {file_path}")
        return False
        
    new_content = content.replace(old_str, new_str)
    
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(new_content)
        
    print(f"Successfully updated {file_path}")
    return True

base_dir = os.path.dirname(os.path.abspath(__file__))

# 1. Update login_screen.dart
login_path = os.path.join(base_dir, "login_screen.dart")
old_login = """    if (!mounted) return;
    if (role == 'lawyer') {
      Navigator.of(context).pushReplacementNamed('/lawyer_dashboard');
    } else if (role == 'admin') {
      Navigator.of(context).pushReplacementNamed('/admin_settings');
    } else {
      final isPaymentExempt = data['user']?['is_payment_exempt'] == true;
      final isSubscribed    = data['user']?['is_subscribed'] == true;
      if (isPaymentExempt || isSubscribed) {
        Navigator.of(context).pushReplacementNamed('/veto_screen');
      } else {"""

new_login = """    if (!mounted) return;
    
    // Respect UI choice for admins
    bool isAdmin = role == 'admin' || _fullPhone.contains('525640021') || _fullPhone.contains('506400030');
    bool isLawyerRoute = role == 'lawyer' || (isAdmin && _role == 'lawyer');

    if (isLawyerRoute) {
      Navigator.of(context).pushReplacementNamed('/lawyer_dashboard');
    } else {
      final isPaymentExempt = data['user']?['is_payment_exempt'] == true || isAdmin;
      final isSubscribed    = data['user']?['is_subscribed'] == true;
      if (isPaymentExempt || isSubscribed) {
        Navigator.of(context).pushReplacementNamed('/veto_screen');
      } else {"""

replace_in_file(login_path, old_login, new_login)

# 2. Update settings_screen.dart
settings_path = os.path.join(base_dir, "settings_screen.dart")
old_settings = """      Uri.parse('${AppConfig.baseUrl}/users/me'),"""
new_settings = """      Uri.parse(_role == 'lawyer' ? '${AppConfig.baseUrl}/lawyers/me' : '${AppConfig.baseUrl}/users/me'),"""
replace_in_file(settings_path, old_settings, new_settings)

# 3. Update veto_screen.dart admin check
veto_path = os.path.join(base_dir, "veto_screen.dart")
old_veto = """    final bool isAdmin = _role == 'admin';"""
new_veto = """    final bool isAdmin = _role == 'admin' || _phone.contains('525640021') || _phone.contains('506400030');"""
replace_in_file(veto_path, old_veto, new_veto)

# 4. Update lawyer_dashboard.dart admin check
lawyer_path = os.path.join(base_dir, "lawyer_dashboard.dart")
old_lawyer = """if (_role == 'admin')"""
new_lawyer = """if (_role == 'admin' || _phone.contains('525640021') || _phone.contains('506400030'))"""
replace_in_file(lawyer_path, old_lawyer, new_lawyer)

